storage/src/multiple_strategy_block_store.rs

```diff
@@ -0,0 +1,109 @@
+// A mixed block store,
+// Stores confirmed blocks in memory
+// Finalized blocks in long term storage of your choice
+// Fetches legacy blocks from faithful
+
+use crate::block_stores::inmemory_block_store::InmemoryBlockStore;
+use async_trait::async_trait;
+use solana_lite_rpc_core::block_storage_interface::{BlockStorageInterface, BlockStorageImpl};
+use solana_rpc_client::nonblocking::rpc_client::RpcClient;
+use solana_rpc_client_api::config::RpcBlockConfig;
+use solana_sdk::{commitment_config::{CommitmentConfig, CommitmentLevel}, slot_history::Slot};
+use solana_transaction_status::{TransactionDetails, UiTransactionEncoding, UiConfirmedBlock};
+use std::sync::{Arc, atomic::{AtomicU64, Ordering}};
+
+pub struct MultipleStrategyBlockStorage {
+    inmemory_for_storage: InmemoryBlockStore, // for confirmed blocks
```

Comment:
In the code, the inmemory_for_storage act has a cache for the persistent_block_storage. I think the persittent_block_storage cache should be implemented in the persistent storage. Usually db has already some cache that can make it useless. You can use LRUCache that can be easier to use and more efficient.

Reply:
The idea of inmemory storage is to store the confirmed block and not finalized blocks. So when blocks are finalized they will be stored persistent storage. In anyways usually most of the clients would ask for recent blocks so some kind of inmemory persistence will be useful.

Reply:
As I understand, blocks have 2 states confirmed and finalized. Does all confirmed block become finalized?
If yes, you can use a LRUCache and add all block to it. By setting the size to 20 I think it's enough to cache all confirmed block. You add all block to the cache and only finalized are written to the DB. This way you always get the confirmed first if the block is in the cache, and you don't have to manage the removing.
If not, the LRUCache will remove them after 20 blocks, so they don't need to be removed.
In my opinion, you don't have to define the interface BlockStorageInterface because the impl don't have the same use case. InmemoryBlockStore is a cache, so it needs in addition: contains and remove (auto remove is not always done by the cache), MultipleStrategyBlockStorage need get and save only and persistent is hidden behind MultipleStrategyBlockStorage. So in the end, only MultipleStrategyBlockStorage should be public.
FYI, I already use this cache: https://crates.io/crates/lru